### PR TITLE
Add support for more queries

### PR DESCRIPTION
**Pull request by seaneking**
@bedeoverend made this into a PR so we can track progress

### CODE REVIEW

File: src/simpla.js [hunk 1]
```diff
@@ -148,6 +165,10 @@ const Simpla = new class Simpla {
     return storeToObserver(this._store).observe(pathInStore, wrappedCallback);
   }
 
+  prefetch(path) {
+    return this.find({ ancestor: path });
```
Comment (seaneking) [inline src/simpla.js]: Prefetch should just return an empty promise that resolves when data is hydrated, not the data itself

**Comment by seaneking** [comment]
Also does this add in support for `type` filtering? Without that `ancestor` is fairly useless

**Comment by seaneking** [comment]
@bedeoverend is there anything blocking this from being merged mainline?

**Comment by bedeoverend** [comment]
Yup so this adds in `type`. 

> is there anything blocking this from being merged mainline?

Yeah, this was just an MVP, there's no live support for `ancestor` and `type` so if someone creates an item that matches it locally, it won't work, same with `observeQuery` won't work.

Also just haven't added any tests for it so generally unstable currently.

**Comment by bedeoverend** [comment]
@seaneking this is pretty much ready to go - realised there was a few query bugs to be patched up. Can you pull down locally and test out? If that's all good then can merge.

**Comment by seaneking** [comment]
👍 LGTM, merging

**Pull request merged**